codemp/rd-rend2/tr_image.cpp

```diff
@@ -2616,16 +2617,22 @@ void R_CreateDiffuseAndSpecMapsFromBaseColorAndRMO(shaderStage_t *stage, const c
 	for (image = hashTable[hash]; image; image = image->next) {
 		if (!strcmp(diffuseName, image->imgName)) {
 			stage->bundle[TB_COLORMAP].image[0] = image;
-			// check for specular map
-			hash = generateHashValue(specularName);
-			for (image = hashTable[hash]; image; image = image->next) {
-				if (!strcmp(specularName, image->imgName)) {
-					stage->bundle[TB_SPECULARMAP].image[0] = image;
-					return;
-				}
-			}
+			foundDiffuse = qtrue;
+			break;
 		}
 	}
+	// check for specular map
+	hash = generateHashValue(specularName);
```

Comment:
Pretty sure you can turn this hash + search loop into a function some how so you can re-use it for the above too. Or maybe there's already a function that does this?